server/internal/server/ollama_client.go

```diff
@@ -1,72 +1,133 @@
 package server
 
 import (
+	"bufio"
+	"bytes"
 	"context"
+	"encoding/json"
+	"fmt"
 	"log"
+	"net/http"
+	"net/url"
+	"os"
 
 	v1 "github.com/llm-operator/inference-manager/api/v1"
-	"github.com/ollama/ollama/api"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-func newClient() (*api.Client, error) {
-	client, err := api.ClientFromEnvironment()
-	if err != nil {
-		return nil, err
+// Modified based on https://github.com/ollama/ollama/blob/main/api/client.go
+// Support OpenAI compatible API, and avoid converting message format conversion.
+
+// client encapsulates client state for interacting with the ollama service.
+type client struct {
+	base *url.URL
+	http *http.Client
+}
+
+func newClient() *client {
+	return &client{
+		base: &url.URL{
+			Scheme: "http",
+			Host:   os.Getenv("OLLAMA_HOST"),
+		},
+		http: http.DefaultClient,
 	}
-	return client, nil
 }
 
-func handleChatRequest(ctx context.Context, req *v1.CreateChatCompletionRequest) (*v1.ChatCompletion, error) {
-	client, err := newClient()
+const maxBufferSize = 512 * 1000
+
+func (c *client) stream(ctx context.Context, method, path string, data any, fn func([]byte) error) error {
+	var buf *bytes.Buffer
+	if data != nil {
```

Comment:
Slightly prefer removing this `nil` check as we always expect a request in this context.

Reply:
any thought?